src/structures/User.js

```diff
@@ -50,6 +41,13 @@ class User extends Base {
       this.username = null;
     }
 
+    /**
+     * Whether or not the user is a bot
+     * @type {?boolean}
+     * @name User#bot
+     */
+    this.bot = !this.partial ? Boolean(data.bot) : null;
```

Comment:
```suggestion
    this.bot = Boolean(data.bot);
```

We should just revert to this and keep it that way. This is too much of a headache, if anything we should just solve this in v13/v14